job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -291,13 +299,39 @@ public void setQuietPeriod(Integer seconds) throws IOException {
     }
 
     @Override public CauseOfBlockage getCauseOfBlockage() {
-        // TODO is there any legitimate reason to block a flow? maybe !isConcurrentBuild()?
+        if (isLogUpdated() && !isConcurrentBuild()) {
```

Comment:
I still suspect <code>isLogUpdated()</code> is not reliable for parallel builds as well as any method using <code>getLastBuild()</code>. If build10 finishes before build9, the result may be errorneous